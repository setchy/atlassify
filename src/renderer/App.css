html,
body,
#root {
  height: 100%;
  -webkit-user-select: none;
}

html::-webkit-scrollbar,
div.flex-grow::-webkit-scrollbar {
  width: 10px;
}

html::-webkit-scrollbar-thumb,
div.flex-grow::-webkit-scrollbar-thumb {
  background-color: #c1c1c1;
  border-radius: 10px;
}

html::-webkit-scrollbar-thumb:hover,
div.flex-grow::-webkit-scrollbar-thumb:hover {
  background-color: #a8a8a8;
  border-radius: 10px;
}

body {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  margin: 0;
  cursor: default;
}

/**
 * Set emoji size according to surrounding text.
 * Ref: https://github.com/jdecked/twemoji?tab=readme-ov-file#inline-styles
 */
img.emoji {
  height: 1em;
  width: 1em;
  margin: 0 0.05em 0 0.1em;
  vertical-align: -0.1em;
}
