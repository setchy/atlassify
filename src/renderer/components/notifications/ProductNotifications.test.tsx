import { act, fireEvent, render, screen } from '@testing-library/react';

import { mockAtlassifyNotifications } from '../../__mocks__/notifications';
import { mockAtlassianCloudAccount, mockSettings } from '../../__mocks__/state';
import { AppContext } from '../../context/App';
import type { ProductName } from '../../types';
import * as comms from '../../utils/comms';
import * as theme from '../../utils/theme';
import { ProductNotifications } from './ProductNotifications';

vi.mock('./NotificationRow', () => ({
  NotificationRow: () => <div>NotificationRow</div>,
}));

const openExternalLinkMock = vi
  .spyOn(comms, 'openExternalLink')
  .mockImplementation(vi.fn());

describe('renderer/components/notifications/ProductNotifications.tsx', () => {
  afterEach(() => {
    vi.clearAllMocks();
  });

  it('should render itself & its children - light mode', () => {
    vi.spyOn(theme, 'isLightMode').mockReturnValue(true);

    const props = {
      account: mockAtlassianCloudAccount,
      product: 'bitbucket' as ProductName,
      productNotifications: mockAtlassifyNotifications,
    };

    const tree = render(
      <AppContext.Provider value={{}}>
        <ProductNotifications {...props} />
      </AppContext.Provider>,
    );

    expect(tree).toMatchSnapshot();
  });

  it('should render itself & its children - dark mode', () => {
    vi.spyOn(theme, 'isLightMode').mockReturnValue(false);

    const props = {
      account: mockAtlassianCloudAccount,
      product: 'bitbucket' as ProductName,
      productNotifications: mockAtlassifyNotifications,
    };

    const tree = render(
      <AppContext.Provider value={{}}>
        <ProductNotifications {...props} />
      </AppContext.Provider>,
    );

    expect(tree).toMatchSnapshot();
  });

  it('should open product home when available', async () => {
    const props = {
      account: mockAtlassianCloudAccount,
      product: 'bitbucket' as ProductName,
      productNotifications: mockAtlassifyNotifications,
    };

    await act(async () => {
      render(<ProductNotifications {...props} />);
    });

    fireEvent.click(screen.getByTestId('product-home'));

    expect(openExternalLinkMock).toHaveBeenCalledTimes(1);
  });

  it('should toggle product notifications visibility', async () => {
    const props = {
      account: mockAtlassianCloudAccount,
      product: 'bitbucket' as ProductName,
      productNotifications: mockAtlassifyNotifications,
    };

    await act(async () => {
      render(<ProductNotifications {...props} />);
    });

    fireEvent.click(screen.getByTestId('product-toggle'));

    const tree = render(<ProductNotifications {...props} />);

    expect(tree).toMatchSnapshot();
  });

  it('should mark all product notifications as read', async () => {
    const props = {
      productNotifications: mockAtlassifyNotifications,
    };

    const markNotificationsReadMock = vi.fn();

    render(
      <AppContext.Provider
        value={{
          settings: mockSettings,
          markNotificationsRead: markNotificationsReadMock,
        }}
      >
        <ProductNotifications {...props} />
      </AppContext.Provider>,
    );

    fireEvent.click(screen.getByTestId('product-mark-as-read'));

    expect(markNotificationsReadMock).toHaveBeenCalledTimes(1);
  });
});
