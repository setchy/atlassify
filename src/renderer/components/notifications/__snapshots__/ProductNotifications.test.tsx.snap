// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer/components/notifications/ProductNotifications.tsx should render itself & its children - dark mode 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1npayr"
      >
        <div
          class="css-15mab0t"
        >
          <div
            class="css-1a0x9v0"
          >
            <div
              role="presentation"
            >
              <button
                class="css-eu6vk3"
                data-testid="product-home"
                type="button"
              >
                <span
                  class="css-1nqby2s"
                >
                  <div
                    class="css-1v6wdk"
                  >
                    <span
                      aria-label="Bitbucket"
                      class="css-5vjqu0"
                      role="img"
                      style="--logo-color: inherit; --logo-fill: currentColor;"
                    >
                      <svg
                        aria-hidden="true"
                        fill="none"
                        focusable="false"
                        height="32"
                        viewBox="0 0 32 32"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        
      
                        <path
                          d="m25.586 15.819-1.618 9.885c-.106.598-.528.95-1.126.95H9.122c-.598 0-1.02-.352-1.126-.95L5.146 8.08c-.105-.598.212-.985.775-.985h20.123c.562 0 .88.387.773.985l-.773 4.644c-.106.668-.493.95-1.126.95H12.816c-.176 0-.282.105-.246.316l.95 5.84c.035.14.14.246.281.246h4.362c.141 0 .246-.105.282-.246l.668-4.222c.07-.527.422-.738.915-.738h4.75c.703 0 .914.351.808.95"
                          fill="#1868DB"
                        />
                        
    
                      </svg>
                    </span>
                    <span
                      class="capitalize"
                    >
                      bitbucket
                    </span>
                    <span
                      class="css-1wkl091"
                    >
                      <span
                        class="css-f9etm4"
                      >
                        2
                      </span>
                    </span>
                  </div>
                   
                </span>
              </button>
            </div>
            <div
              class="css-gokkc4"
            >
              <div
                role="presentation"
              >
                <button
                  class="css-1vyxgfe"
                  data-testid="product-mark-as-read"
                  type="button"
                >
                  <span
                    class="css-1spmf3f"
                  >
                    <svg
                      aria-label="Unread Indicator"
                      height="24"
                      role="img"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                      xmlns:xlink="http://www.w3.org/1999/xlink"
                    >
                      <circle
                        cx="50%"
                        cy="50%"
                        fill="var(--ds-text)"
                        r="4"
                      />
                    </svg>
                    <span
                      class="css-b5o75w"
                    >
                      Mark all product notifications as read
                    </span>
                  </span>
                </button>
              </div>
              <div
                role="presentation"
              >
                <button
                  class="css-1vyxgfe"
                  data-testid="product-toggle"
                  type="button"
                >
                  <span
                    class="css-1spmf3f"
                  >
                    <span
                      aria-hidden="true"
                      class="css-9693te"
                      style="color: currentColor;"
                    >
                      <svg
                        class="css-1a5x76h"
                        fill="none"
                        role="presentation"
                        viewBox="0 0 12 12"
                      >
                        <path
                          d="M10.5 4.5 8.25 6.75 6 9 3.75 6.75 1.5 4.5"
                          stroke="currentcolor"
                          stroke-linejoin="round"
                          stroke-width="1.5"
                        />
                      </svg>
                    </span>
                    <span
                      class="css-b5o75w"
                    >
                      Hide product notifications
                    </span>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div>
          NotificationRow
        </div>
        <div>
          NotificationRow
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1npayr"
    >
      <div
        class="css-15mab0t"
      >
        <div
          class="css-1a0x9v0"
        >
          <div
            role="presentation"
          >
            <button
              class="css-eu6vk3"
              data-testid="product-home"
              type="button"
            >
              <span
                class="css-1nqby2s"
              >
                <div
                  class="css-1v6wdk"
                >
                  <span
                    aria-label="Bitbucket"
                    class="css-5vjqu0"
                    role="img"
                    style="--logo-color: inherit; --logo-fill: currentColor;"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      focusable="false"
                      height="32"
                      viewBox="0 0 32 32"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      
      
                      <path
                        d="m25.586 15.819-1.618 9.885c-.106.598-.528.95-1.126.95H9.122c-.598 0-1.02-.352-1.126-.95L5.146 8.08c-.105-.598.212-.985.775-.985h20.123c.562 0 .88.387.773.985l-.773 4.644c-.106.668-.493.95-1.126.95H12.816c-.176 0-.282.105-.246.316l.95 5.84c.035.14.14.246.281.246h4.362c.141 0 .246-.105.282-.246l.668-4.222c.07-.527.422-.738.915-.738h4.75c.703 0 .914.351.808.95"
                        fill="#1868DB"
                      />
                      
    
                    </svg>
                  </span>
                  <span
                    class="capitalize"
                  >
                    bitbucket
                  </span>
                  <span
                    class="css-1wkl091"
                  >
                    <span
                      class="css-f9etm4"
                    >
                      2
                    </span>
                  </span>
                </div>
                 
              </span>
            </button>
          </div>
          <div
            class="css-gokkc4"
          >
            <div
              role="presentation"
            >
              <button
                class="css-1vyxgfe"
                data-testid="product-mark-as-read"
                type="button"
              >
                <span
                  class="css-1spmf3f"
                >
                  <svg
                    aria-label="Unread Indicator"
                    height="24"
                    role="img"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                    xmlns:xlink="http://www.w3.org/1999/xlink"
                  >
                    <circle
                      cx="50%"
                      cy="50%"
                      fill="var(--ds-text)"
                      r="4"
                    />
                  </svg>
                  <span
                    class="css-b5o75w"
                  >
                    Mark all product notifications as read
                  </span>
                </span>
              </button>
            </div>
            <div
              role="presentation"
            >
              <button
                class="css-1vyxgfe"
                data-testid="product-toggle"
                type="button"
              >
                <span
                  class="css-1spmf3f"
                >
                  <span
                    aria-hidden="true"
                    class="css-9693te"
                    style="color: currentColor;"
                  >
                    <svg
                      class="css-1a5x76h"
                      fill="none"
                      role="presentation"
                      viewBox="0 0 12 12"
                    >
                      <path
                        d="M10.5 4.5 8.25 6.75 6 9 3.75 6.75 1.5 4.5"
                        stroke="currentcolor"
                        stroke-linejoin="round"
                        stroke-width="1.5"
                      />
                    </svg>
                  </span>
                  <span
                    class="css-b5o75w"
                  >
                    Hide product notifications
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div>
        NotificationRow
      </div>
      <div>
        NotificationRow
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renderer/components/notifications/ProductNotifications.tsx should render itself & its children - light mode 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1npayr"
      >
        <div
          class="css-ea7a9f"
        >
          <div
            class="css-1a0x9v0"
          >
            <div
              role="presentation"
            >
              <button
                class="css-eu6vk3"
                data-testid="product-home"
                type="button"
              >
                <span
                  class="css-1nqby2s"
                >
                  <div
                    class="css-1v6wdk"
                  >
                    <span
                      aria-label="Bitbucket"
                      class="css-5vjqu0"
                      role="img"
                      style="--logo-color: inherit; --logo-fill: currentColor;"
                    >
                      <svg
                        aria-hidden="true"
                        fill="none"
                        focusable="false"
                        height="32"
                        viewBox="0 0 32 32"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        
      
                        <path
                          d="m25.586 15.819-1.618 9.885c-.106.598-.528.95-1.126.95H9.122c-.598 0-1.02-.352-1.126-.95L5.146 8.08c-.105-.598.212-.985.775-.985h20.123c.562 0 .88.387.773.985l-.773 4.644c-.106.668-.493.95-1.126.95H12.816c-.176 0-.282.105-.246.316l.95 5.84c.035.14.14.246.281.246h4.362c.141 0 .246-.105.282-.246l.668-4.222c.07-.527.422-.738.915-.738h4.75c.703 0 .914.351.808.95"
                          fill="#1868DB"
                        />
                        
    
                      </svg>
                    </span>
                    <span
                      class="capitalize"
                    >
                      bitbucket
                    </span>
                    <span
                      class="css-1wkl091"
                    >
                      <span
                        class="css-f9etm4"
                      >
                        2
                      </span>
                    </span>
                  </div>
                   
                </span>
              </button>
            </div>
            <div
              class="css-gokkc4"
            >
              <div
                role="presentation"
              >
                <button
                  class="css-1vyxgfe"
                  data-testid="product-mark-as-read"
                  type="button"
                >
                  <span
                    class="css-1spmf3f"
                  >
                    <svg
                      aria-label="Unread Indicator"
                      height="24"
                      role="img"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                      xmlns:xlink="http://www.w3.org/1999/xlink"
                    >
                      <circle
                        cx="50%"
                        cy="50%"
                        fill="var(--ds-text)"
                        r="4"
                      />
                    </svg>
                    <span
                      class="css-b5o75w"
                    >
                      Mark all product notifications as read
                    </span>
                  </span>
                </button>
              </div>
              <div
                role="presentation"
              >
                <button
                  class="css-1vyxgfe"
                  data-testid="product-toggle"
                  type="button"
                >
                  <span
                    class="css-1spmf3f"
                  >
                    <span
                      aria-hidden="true"
                      class="css-9693te"
                      style="color: currentColor;"
                    >
                      <svg
                        class="css-1a5x76h"
                        fill="none"
                        role="presentation"
                        viewBox="0 0 12 12"
                      >
                        <path
                          d="M10.5 4.5 8.25 6.75 6 9 3.75 6.75 1.5 4.5"
                          stroke="currentcolor"
                          stroke-linejoin="round"
                          stroke-width="1.5"
                        />
                      </svg>
                    </span>
                    <span
                      class="css-b5o75w"
                    >
                      Hide product notifications
                    </span>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div>
          NotificationRow
        </div>
        <div>
          NotificationRow
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1npayr"
    >
      <div
        class="css-ea7a9f"
      >
        <div
          class="css-1a0x9v0"
        >
          <div
            role="presentation"
          >
            <button
              class="css-eu6vk3"
              data-testid="product-home"
              type="button"
            >
              <span
                class="css-1nqby2s"
              >
                <div
                  class="css-1v6wdk"
                >
                  <span
                    aria-label="Bitbucket"
                    class="css-5vjqu0"
                    role="img"
                    style="--logo-color: inherit; --logo-fill: currentColor;"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      focusable="false"
                      height="32"
                      viewBox="0 0 32 32"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      
      
                      <path
                        d="m25.586 15.819-1.618 9.885c-.106.598-.528.95-1.126.95H9.122c-.598 0-1.02-.352-1.126-.95L5.146 8.08c-.105-.598.212-.985.775-.985h20.123c.562 0 .88.387.773.985l-.773 4.644c-.106.668-.493.95-1.126.95H12.816c-.176 0-.282.105-.246.316l.95 5.84c.035.14.14.246.281.246h4.362c.141 0 .246-.105.282-.246l.668-4.222c.07-.527.422-.738.915-.738h4.75c.703 0 .914.351.808.95"
                        fill="#1868DB"
                      />
                      
    
                    </svg>
                  </span>
                  <span
                    class="capitalize"
                  >
                    bitbucket
                  </span>
                  <span
                    class="css-1wkl091"
                  >
                    <span
                      class="css-f9etm4"
                    >
                      2
                    </span>
                  </span>
                </div>
                 
              </span>
            </button>
          </div>
          <div
            class="css-gokkc4"
          >
            <div
              role="presentation"
            >
              <button
                class="css-1vyxgfe"
                data-testid="product-mark-as-read"
                type="button"
              >
                <span
                  class="css-1spmf3f"
                >
                  <svg
                    aria-label="Unread Indicator"
                    height="24"
                    role="img"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                    xmlns:xlink="http://www.w3.org/1999/xlink"
                  >
                    <circle
                      cx="50%"
                      cy="50%"
                      fill="var(--ds-text)"
                      r="4"
                    />
                  </svg>
                  <span
                    class="css-b5o75w"
                  >
                    Mark all product notifications as read
                  </span>
                </span>
              </button>
            </div>
            <div
              role="presentation"
            >
              <button
                class="css-1vyxgfe"
                data-testid="product-toggle"
                type="button"
              >
                <span
                  class="css-1spmf3f"
                >
                  <span
                    aria-hidden="true"
                    class="css-9693te"
                    style="color: currentColor;"
                  >
                    <svg
                      class="css-1a5x76h"
                      fill="none"
                      role="presentation"
                      viewBox="0 0 12 12"
                    >
                      <path
                        d="M10.5 4.5 8.25 6.75 6 9 3.75 6.75 1.5 4.5"
                        stroke="currentcolor"
                        stroke-linejoin="round"
                        stroke-width="1.5"
                      />
                    </svg>
                  </span>
                  <span
                    class="css-b5o75w"
                  >
                    Hide product notifications
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div>
        NotificationRow
      </div>
      <div>
        NotificationRow
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renderer/components/notifications/ProductNotifications.tsx should toggle product notifications visibility 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1npayr"
      >
        <div
          class="css-15mab0t"
        >
          <div
            class="css-1a0x9v0"
          >
            <div
              role="presentation"
            >
              <button
                class="css-eu6vk3"
                data-testid="product-home"
                type="button"
              >
                <span
                  class="css-1nqby2s"
                >
                  <div
                    class="css-1v6wdk"
                  >
                    <span
                      aria-label="Bitbucket"
                      class="css-5vjqu0"
                      role="img"
                      style="--logo-color: inherit; --logo-fill: currentColor;"
                    >
                      <svg
                        aria-hidden="true"
                        fill="none"
                        focusable="false"
                        height="32"
                        viewBox="0 0 32 32"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        
      
                        <path
                          d="m25.586 15.819-1.618 9.885c-.106.598-.528.95-1.126.95H9.122c-.598 0-1.02-.352-1.126-.95L5.146 8.08c-.105-.598.212-.985.775-.985h20.123c.562 0 .88.387.773.985l-.773 4.644c-.106.668-.493.95-1.126.95H12.816c-.176 0-.282.105-.246.316l.95 5.84c.035.14.14.246.281.246h4.362c.141 0 .246-.105.282-.246l.668-4.222c.07-.527.422-.738.915-.738h4.75c.703 0 .914.351.808.95"
                          fill="#1868DB"
                        />
                        
    
                      </svg>
                    </span>
                    <span
                      class="capitalize"
                    >
                      bitbucket
                    </span>
                    <span
                      class="css-1wkl091"
                    >
                      <span
                        class="css-f9etm4"
                      >
                        2
                      </span>
                    </span>
                  </div>
                   
                </span>
              </button>
            </div>
            <div
              class="css-gokkc4"
            >
              <div
                role="presentation"
              >
                <button
                  class="css-1vyxgfe"
                  data-testid="product-mark-as-read"
                  type="button"
                >
                  <span
                    class="css-1spmf3f"
                  >
                    <svg
                      aria-label="Unread Indicator"
                      height="24"
                      role="img"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                      xmlns:xlink="http://www.w3.org/1999/xlink"
                    >
                      <circle
                        cx="50%"
                        cy="50%"
                        fill="var(--ds-text)"
                        r="4"
                      />
                    </svg>
                    <span
                      class="css-b5o75w"
                    >
                      Mark all product notifications as read
                    </span>
                  </span>
                </button>
              </div>
              <div
                role="presentation"
              >
                <button
                  class="css-1vyxgfe"
                  data-testid="product-toggle"
                  type="button"
                >
                  <span
                    class="css-1spmf3f"
                  >
                    <span
                      aria-hidden="true"
                      class="css-9693te"
                      style="color: currentColor;"
                    >
                      <svg
                        class="css-1a5x76h"
                        fill="none"
                        role="presentation"
                        viewBox="0 0 12 12"
                      >
                        <path
                          d="M4.5 10.5 9 6 4.5 1.5"
                          stroke="currentcolor"
                          stroke-linejoin="round"
                          stroke-width="1.5"
                        />
                      </svg>
                    </span>
                    <span
                      class="css-b5o75w"
                    >
                      Show product notifications
                    </span>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="css-1npayr"
      >
        <div
          class="css-15mab0t"
        >
          <div
            class="css-1a0x9v0"
          >
            <div
              role="presentation"
            >
              <button
                class="css-eu6vk3"
                data-testid="product-home"
                type="button"
              >
                <span
                  class="css-1nqby2s"
                >
                  <div
                    class="css-1v6wdk"
                  >
                    <span
                      aria-label="Bitbucket"
                      class="css-5vjqu0"
                      role="img"
                      style="--logo-color: inherit; --logo-fill: currentColor;"
                    >
                      <svg
                        aria-hidden="true"
                        fill="none"
                        focusable="false"
                        height="32"
                        viewBox="0 0 32 32"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        
      
                        <path
                          d="m25.586 15.819-1.618 9.885c-.106.598-.528.95-1.126.95H9.122c-.598 0-1.02-.352-1.126-.95L5.146 8.08c-.105-.598.212-.985.775-.985h20.123c.562 0 .88.387.773.985l-.773 4.644c-.106.668-.493.95-1.126.95H12.816c-.176 0-.282.105-.246.316l.95 5.84c.035.14.14.246.281.246h4.362c.141 0 .246-.105.282-.246l.668-4.222c.07-.527.422-.738.915-.738h4.75c.703 0 .914.351.808.95"
                          fill="#1868DB"
                        />
                        
    
                      </svg>
                    </span>
                    <span
                      class="capitalize"
                    >
                      bitbucket
                    </span>
                    <span
                      class="css-1wkl091"
                    >
                      <span
                        class="css-f9etm4"
                      >
                        2
                      </span>
                    </span>
                  </div>
                   
                </span>
              </button>
            </div>
            <div
              class="css-gokkc4"
            >
              <div
                role="presentation"
              >
                <button
                  class="css-1vyxgfe"
                  data-testid="product-mark-as-read"
                  type="button"
                >
                  <span
                    class="css-1spmf3f"
                  >
                    <svg
                      aria-label="Unread Indicator"
                      height="24"
                      role="img"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                      xmlns:xlink="http://www.w3.org/1999/xlink"
                    >
                      <circle
                        cx="50%"
                        cy="50%"
                        fill="var(--ds-text)"
                        r="4"
                      />
                    </svg>
                    <span
                      class="css-b5o75w"
                    >
                      Mark all product notifications as read
                    </span>
                  </span>
                </button>
              </div>
              <div
                role="presentation"
              >
                <button
                  class="css-1vyxgfe"
                  data-testid="product-toggle"
                  type="button"
                >
                  <span
                    class="css-1spmf3f"
                  >
                    <span
                      aria-hidden="true"
                      class="css-9693te"
                      style="color: currentColor;"
                    >
                      <svg
                        class="css-1a5x76h"
                        fill="none"
                        role="presentation"
                        viewBox="0 0 12 12"
                      >
                        <path
                          d="M10.5 4.5 8.25 6.75 6 9 3.75 6.75 1.5 4.5"
                          stroke="currentcolor"
                          stroke-linejoin="round"
                          stroke-width="1.5"
                        />
                      </svg>
                    </span>
                    <span
                      class="css-b5o75w"
                    >
                      Hide product notifications
                    </span>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div>
          NotificationRow
        </div>
        <div>
          NotificationRow
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1npayr"
    >
      <div
        class="css-15mab0t"
      >
        <div
          class="css-1a0x9v0"
        >
          <div
            role="presentation"
          >
            <button
              class="css-eu6vk3"
              data-testid="product-home"
              type="button"
            >
              <span
                class="css-1nqby2s"
              >
                <div
                  class="css-1v6wdk"
                >
                  <span
                    aria-label="Bitbucket"
                    class="css-5vjqu0"
                    role="img"
                    style="--logo-color: inherit; --logo-fill: currentColor;"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      focusable="false"
                      height="32"
                      viewBox="0 0 32 32"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      
      
                      <path
                        d="m25.586 15.819-1.618 9.885c-.106.598-.528.95-1.126.95H9.122c-.598 0-1.02-.352-1.126-.95L5.146 8.08c-.105-.598.212-.985.775-.985h20.123c.562 0 .88.387.773.985l-.773 4.644c-.106.668-.493.95-1.126.95H12.816c-.176 0-.282.105-.246.316l.95 5.84c.035.14.14.246.281.246h4.362c.141 0 .246-.105.282-.246l.668-4.222c.07-.527.422-.738.915-.738h4.75c.703 0 .914.351.808.95"
                        fill="#1868DB"
                      />
                      
    
                    </svg>
                  </span>
                  <span
                    class="capitalize"
                  >
                    bitbucket
                  </span>
                  <span
                    class="css-1wkl091"
                  >
                    <span
                      class="css-f9etm4"
                    >
                      2
                    </span>
                  </span>
                </div>
                 
              </span>
            </button>
          </div>
          <div
            class="css-gokkc4"
          >
            <div
              role="presentation"
            >
              <button
                class="css-1vyxgfe"
                data-testid="product-mark-as-read"
                type="button"
              >
                <span
                  class="css-1spmf3f"
                >
                  <svg
                    aria-label="Unread Indicator"
                    height="24"
                    role="img"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                    xmlns:xlink="http://www.w3.org/1999/xlink"
                  >
                    <circle
                      cx="50%"
                      cy="50%"
                      fill="var(--ds-text)"
                      r="4"
                    />
                  </svg>
                  <span
                    class="css-b5o75w"
                  >
                    Mark all product notifications as read
                  </span>
                </span>
              </button>
            </div>
            <div
              role="presentation"
            >
              <button
                class="css-1vyxgfe"
                data-testid="product-toggle"
                type="button"
              >
                <span
                  class="css-1spmf3f"
                >
                  <span
                    aria-hidden="true"
                    class="css-9693te"
                    style="color: currentColor;"
                  >
                    <svg
                      class="css-1a5x76h"
                      fill="none"
                      role="presentation"
                      viewBox="0 0 12 12"
                    >
                      <path
                        d="M10.5 4.5 8.25 6.75 6 9 3.75 6.75 1.5 4.5"
                        stroke="currentcolor"
                        stroke-linejoin="round"
                        stroke-width="1.5"
                      />
                    </svg>
                  </span>
                  <span
                    class="css-b5o75w"
                  >
                    Hide product notifications
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div>
        NotificationRow
      </div>
      <div>
        NotificationRow
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
